go/tools/go-upgrade/go-upgrade.go

```diff
@@ -251,23 +257,41 @@ func currentGolangVersion() (*version.Version, error) {
 	return version.NewVersion(versionStr[1])
 }
 
-func currentBootstrapVersion() (float64, error) {
+func currentBootstrapVersion() (bootstrapVersion, error) {
 	contentRaw, err := os.ReadFile("Makefile")
 	if err != nil {
-		return 0, err
+		return bootstrapVersion{}, err
 	}
 	content := string(contentRaw)
 
 	versre := regexp.MustCompile("(?i).*BOOTSTRAP_VERSION[[:space:]]*=[[:space:]]*([0-9.]+).*")
 	versionStr := versre.FindStringSubmatch(content)
 	if len(versionStr) != 2 {
-		return 0, fmt.Errorf("malformatted error, got: %v", versionStr)
+		return bootstrapVersion{}, fmt.Errorf("malformatted error, got: %v", versionStr)
+	}
+	_, err = strconv.ParseFloat(versionStr[1], 64)
+	if err != nil {
+		return bootstrapVersion{}, err
```

Comment:
@frouioui I don't think we need this check? We already detect errors if we try to parse the separate parts anyway?

Reply:
I merged the PR to avoid having to wait for CI all over again. Let me fix this in the backports and in a subsequent `main` PR.